### PR TITLE
ceph-osd: some flags are not documented in the help output

### PR DESCRIPTION
**Pull request by gmayyyha**
Fixes: http://tracker.ceph.com/issues/20057

Signed-off-by: Yanhu Cao <gmayyyha@gmail.com>

### CODE REVIEW

**Comment by gmayyyha** [comment]
retest this please

**Comment by gmayyyha** [comment]
retest this please

File: src/ceph_osd.cc [hunk 1]
```diff
@@ -73,6 +73,10 @@ static void usage()
        << "  --osd-journal PATH\n"
        << "                    journal file or block device\n"
        << "  --mkfs            create a [new] data directory\n"
+       << "  --mkkey           generate a new secret key\n"
+       << "  --monmap          specify the path to the monitor map\n"
+       << "  --osd-uuid        specify the OSD's fsid\n"
+       << "  --keyring         specify a path to the osd keyring\n"
```
Comment (tchaikov) [inline src/ceph_osd.cc]: please see @gregsfortytwo 's comment at http://tracker.ceph.com/issues/20057#note-8, except for `--mkkey`, all these options  are parsed by `md_config_t::parse_argv()`, which checks the args against **all** config_opts.  so, we can override all options by passing it to `ceph-osd`, like 
```
ceph-osd --osd-pool-default-pg-num
```

but shall we document them all also? i don't really think so.

Comment (liewegas) [inline src/ceph_osd.cc]: These are specifically relevant to --mkfs, so it seems ok to manually include them here?

Comment (tchaikov) [inline src/ceph_osd.cc]: okay, makes sense to me!

**Comment by gmayyyha** [comment]
@liewegas master branch has `--mkkey` option, the 3 options do not exist. need to document these options in the master branch?

**Comment by gmayyyha** [comment]
retest this please

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-09-01_21:29:08-rados-wip-sage-testing-2017-09-01-1033-distro-basic-smithi/

**Pull request merged**